File: backend/engine/replies.py
"""Deterministic Hinglish reply templates.

Faithful 1:1 port of app/lib/engine/replies.ts. Two renderings per reply: ``text``
(display) and ``speech`` (TTS-friendly).
"""
from __future__ import annotations

import re

MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August",
          "September", "October", "November", "December"]


def spoken_date(iso: str | None) -> str:
    if not iso:
        return ""
    m = re.match(r"^(\d{4})-(\d{2})-(\d{2})", iso)
    return f"{int(m.group(3))} {MONTHS[int(m.group(2)) - 1]}" if m else iso


ATTR_HI = {
    "rebar_spacing": "rebar spacing", "stirrup_spacing": "stirrup spacing", "cover": "cover",
    "thickness": "thickness", "rebar_dia": "bar dia", "size": "size", "grade": "grade",
    "bar_count": "bar count",
}


def attr_label(a: str | None) -> str:
    if not a:
        return ""
    return ATTR_HI.get(a) or a.replace("_", " ")


ACT_HI = {"hot_work": "hot work", "pour": "dhalai", "height": "height work", "excavation": "khudai"}


def level_label(l: str | None) -> str:
    if not l:
        return ""
    return re.sub(r"^L(\d+)$", r"Level \1", l)


def _v(s: dict, k: str):
    sl = s.get(k)
    return sl["value"] if sl else None


def loc_label(c, s: dict) -> str:
    loc = c.location
    if loc:
        seg = loc["location_id"].split(":")[1] if ":" in loc["location_id"] else loc["location_id"]
        if (loc.get("grid") or seg) == "Slab":
            return f"{level_label(loc.get('level'))} slab"
        head = loc.get("grid") or loc.get("zone") or seg
        head = re.sub(r"^Zone([A-Z])$", r"Zone \1", head)
        return f"{head}, {level_label(loc.get('level'))}"
    return ", ".join(x for x in [_v(s, "grid") or _v(s, "zone"), level_label(_v(s, "level"))] if x)


def to_speech(t: str) -> str:
    t = re.sub(r"(\d+(?:\.\d+)?)\s?mm\b", r"\1 millimeter", t)
    t = re.sub(r"±\s?(\d+)", r"plus minus \1", t)
    t = re.sub(r"\bRFI-0*(\d+)", lambda m: f"R F I {m.group(1)}", t)
    t = re.sub(r"\bNCR\b", "N C R", t)
    t = re.sub(r"\bOBS-0*(\d+)", lambda m: f"observation number {m.group(1)}", t)
    t = re.sub(r"\bHWP-0*(\d+)", lambda m: f"H W P {m.group(1)}", t)
    t = re.sub(r"\b([A-Z])-(\d{3})\b", r"\1 \2", t)
    t = re.sub(r"\b([A-H])-(\d{1,2})\b", r"\1 \2", t)
    t = re.sub(r"\bR(\d+)\b", r"revision \1", t)
    t = re.sub(r"\bL(\d+)\b", r"level \1", t)
    t = re.sub(r"QC-[A-Z]+-[A-Z]+-\d+", "checklist", t)
    t = re.sub(r"\bCL-[A-Z0-9-]+", "", t)
    t = re.sub(r"—", ", ", t)
    t = re.sub(r"\s+", " ", t)
    return t.strip()


def R(text: str) -> dict:
    return {"text": text, "speech": to_speech(text)}


reply = R


def challenge_reply(c, s: dict, prefix: str = "") -> dict:
    parts: list[str] = []
    if prefix:
        parts.append(prefix)
    loc = loc_label(c, s)
    attr = attr_label(c.attribute or _v(s, "attribute"))
    for k in c.contradictions:
        parts.append(_describe(k, c, s, loc, attr))
    has_dim = any(k["kind"] == "dimension_mismatch" for k in c.contradictions)
    parts.append(
        "Kya aap observation log karna chahte hain, RFI raise karna chahte hain, ya NCR?"
        if has_dim else
        "Kya aap observation log karna chahte hain, ya RFI raise karna chahte hain?"
    )
    return R(" ".join(parts))


def _describe(k: dict, c, s: dict, loc: str, attr: str) -> str:
    e = k["evidence"]
    kind = k["kind"]
    if kind == "revision_mismatch":
        t = (f"Drawing {e.get('drawing_number')} ki latest revision {e.get('revision')} hai, "
             f"jo {spoken_date(e.get('issued_on'))} ko issue hui thi.")
        if e.get("expected") is not None:
            t += f" Usme {loc} ki {attr} {e['expected']} {e.get('unit') or 'mm'} hai."
        if e.get("via_rfi"):
            t += f" {e['via_rfi']} ne yeh confirm kiya tha."
        if e.get("claimed_revision"):
            t += (f" {e['claimed_revision']} superseded hai"
                  + (f", usme {e['claimed_revision_value']} tha"
                     if e.get("claimed_revision_value") is not None else "") + ".")
        if (c.value is not None and e.get("expected") is not None
                and not any(x["kind"] == "dimension_mismatch" for x in c.contradictions)):
            t += f" Aapka {c.value} {c.unit or 'mm'} latest drawing se match karta hai."
        return t
    if kind == "dimension_mismatch":
        if e.get("claimed") is not None and c.value is not None and e["claimed"] == c.value:
            return (f"Dhyan dijiye: {loc} pe aapne {attr} {c.value} {e.get('unit') or 'mm'} bola, "
                    f"lekin {e.get('drawing_number')} {e.get('revision')} mein {e.get('expected')} "
                    f"{e.get('unit') or 'mm'} hai, tolerance ±{e.get('tolerance') or 0}. "
                    f"Yeh {abs(e['claimed'] - e['expected'])} {e.get('unit') or 'mm'} ka farak hai."
                    + (f" Reference {e['code_ref']}." if e.get("code_ref") else ""))
        return (f"Aapne drawing mein {e.get('claimed')} bataya, lekin {e.get('drawing_number')} "
                f"{e.get('revision')} mein {e.get('expected')} {e.get('unit') or 'mm'} hai.")
    if kind == "unknown_drawing":
        if e.get("drawing_id"):
            return (f"Drawing {e.get('drawing_number')} is location ke liye applicable nahi hai; "
                    f"{e['drawing_id'].replace('@', ' ')} lagu hai, issue {spoken_date(e.get('issued_on'))}.")
        return f"Drawing {e.get('drawing_number')} register mein nahi mili."
    return k["detail"]


def unknown_drawing_reply(k: dict) -> dict:
    cands = ", ".join((k["evidence"].get("candidates") or [])[:4])
    return R(f"Drawing {k['evidence'].get('drawing_number')} register mein nahi mili. "
            f"Kripya drawing number dobara bataiye"
            + (f" — register mein {cands} hain" if cands else "")
            + ". Tab tak kuch log nahi hoga.")


def missing_reply(c, s: dict) -> dict:
    m = c.missing[0]
    f = m["evidence"].get("field")
    if f == "level":
        levels = " ya ".join(level_label(i.split(":")[2]) for i in (m["evidence"].get("candidates") or []))
        return R(f"{_v(s, 'grid') or _v(s, 'zone')} kaunse level pe hai — {levels}?")
    if f == "element":
        return R(f"{loc_label(c, s)} pe kaunsa element — {' ya '.join(m['evidence'].get('candidates') or [])}?")
    if f == "location_unknown":
        return R(f"{m['evidence'].get('claimed')} project register mein nahi mila. Grid line dobara bataiye.")
    if f == "location":
        return R("Location bataiye — kaunsi grid line aur level? Jaise C-5, Level 3.")
    if f == "attribute":
        return R(f"{loc_label(c, s) or 'Wahan'} pe kya check kiya — spacing, cover ya thickness?")
    if f == "value":
        return R(f"{attr_label(c.attribute or _v(s, 'attribute')) or 'Value'} kitni hai? mm mein bataiye.")
    return R("Thoda aur detail chahiye. Location, attribute aur value bataiye.")


_SLOT_SAY = {
    "grid": lambda s: f"grid {_v(s, 'grid')}",
    "level": lambda s: level_label(_v(s, "level")),
    "zone": lambda s: f"{_v(s, 'zone')}",
    "attribute": lambda s: attr_label(_v(s, "attribute")),
    "value": lambda s: f"{_v(s, 'value')} {_v(s, 'unit') or 'mm'}",
    "drawingNumber": lambda s: f"drawing {_v(s, 'drawingNumber')}",
    "element": lambda s: f"{_v(s, 'element')}",
    "activity": lambda s: ACT_HI.get(_v(s, "activity") or "") or str(_v(s, "activity")),
}


def confirm_slots_reply(names: list[str], s: dict, resolved_drawing: str | None = None) -> dict:
    said = ", ".join(
        (f"drawing {resolved_drawing}" if (n == "drawingNumber" and resolved_drawing)
         else (_SLOT_SAY[n](s) if n in _SLOT_SAY else n))
        for n in names
    )
    return R(f"Awaaz saaf nahi thi. Maine suna: {said}. Kya yeh sahi hai? Haan ya nahi boliye.")


def _short_list(items: list[str] | None) -> str:
    if not items:
        return ""
    short = [re.split(r"[,(—;]| - ", i)[0].strip() for i in items]
    head = "; ".join(short[:2])
    return f"{head}, aur {len(items) - 2} aur items" if len(items) > 2 else head


def blocker_reply(c, s: dict, attempted: str | None = None) -> dict:
    parts: list[str] = []
    if attempted:
        parts.append(f"{attempted} abhi allowed nahi hai.")
    parts.append("Ruko.")
    loc = loc_label(c, s)
    for b in c.blockers:
        if b["kind"] == "permit_blocker":
            act_lbl = ACT_HI.get(_v(s, "activity") or "") or "kaam"
            if b["evidence"].get("permit_id"):
                parts.append(
                    f"{loc} pe {act_lbl} permit {b['evidence']['permit_id']} active hai, lekin "
                    f"mandatory check pending hai: {_short_list(b['evidence'].get('checks'))}. "
                    f"Jab tak yeh confirm na ho, {act_lbl} shuru nahi ho sakta."
                )
            else:
                parts.append(
                    f"{loc} pe {ACT_HI.get(_v(s, 'activity') or '') or 'is kaam'} ke liye koi active permit nahi hai."
                )
        elif b["kind"] == "hold_point_blocker":
            parts.append(
                f"{loc} ka pre-pour checklist {b['evidence'].get('checklist')} ka hold point release nahi hua: "
                f"{_short_list(b['evidence'].get('items'))}. Dhalai shuru nahi ho sakti."
            )
    parts.append("Kaam rokna hai, NCR raise karna hai, ya cancel?")
    return R(" ".join(parts))


def readback_reply(c, s: dict, prefix: str = "") -> dict:
    loc = loc_label(c, s)
    d = f"{c.drawing['drawing_number']} {c.drawing['revision']}" if c.drawing else None
    t = (prefix + " ") if prefix else ""
    if c.mode == "defect":
        t += (f"{loc} pe {_v(s, 'element') or ''} {_v(s, 'defect')}"
              + (f", drawing {d} se link hoga" if d else "") + ".")
    elif c.mode == "activity":
        t += f"{loc} pe {ACT_HI.get(_v(s, 'activity') or '')} ke liye koi blocker nahi mila."
    else:
        t += (f"{loc}, {c.element or _v(s, 'element') or ''} {attr_label(c.attribute)} {c.value} "
              f"{c.unit or 'mm'}"
              + (f" — {d} ke hisaab se sahi hai" if (c.fact and d)
                 else (f" — drawing {d}" if d else "")) + ".")
    t = re.sub(r"\s+", " ", t).replace(" ,", ",")
    return R(f"{t} Log kar doon?")


def done_reply(decision: str, obs_id: str | None = None, rfi_id: str | None = None,
               drawing_label: str | None = None) -> dict:
    if decision == "log_observation":
        return R(f"Observation {obs_id} log ho gaya"
                 + (f", drawing {drawing_label} ke saath link kiya" if drawing_label else "") + ".")
    if decision == "raise_rfi":
        return R(f"{rfi_id} raise ho gaya, status Open. Observation {obs_id} usse link kar diya.")
    if decision == "raise_ncr":
        return R(f"NCR ke liye observation {obs_id} log ho gaya. QA team ko notify kijiye.")
    if decision == "stop_work":
        return R(f"Kaam rokne ka order {obs_id} log ho gaya. Safety officer ko turant inform kijiye.")
    if decision == "cancel":
        return R("Theek hai, cancel kar diya. Kuch log nahi hua.")
    return R("Theek hai.")


REFUSE = {
    "confirmFirst": "Pehle confirm kijiye, phir log karunga.",
    "clarifyFirst": "Log karne se pehle ek baat:",
    "ambiguousYes": "Log karna hai ya RFI raise karna hai? Saaf boliye.",
    "rfiDeclined": "Theek hai, RFI nahi. Toh observation log kar doon?",
    "nothing": "Main sun raha hoon. Location, attribute aur value boliye — jaise, C-5 pe spacing 180 mm.",
}

DECISION_LABEL = {
    "log_observation": "Observation log karna",
    "raise_rfi": "RFI raise karna",
    "raise_ncr": "NCR raise karna",
    "stop_work": "Kaam rokna",
    "cancel": "Cancel",
}

File: app/lib/engine/replies.ts
// Deterministic Hinglish reply templates. Two renderings per reply: `text` (display) and `speech`
// (TTS-friendly: units spelled, IDs spaced, dates spoken).
import type { CheckResult, Contradiction, SlotName, Slots } from "./contradictions";

const MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
export function spokenDate(iso?: string | null): string {
  if (!iso) return "";
  const m = /^(\d{4})-(\d{2})-(\d{2})/.exec(iso);
  return m ? `${Number(m[3])} ${MONTHS[Number(m[2]) - 1]}` : iso;
}
const ATTR_HI: Record<string, string> = {
  rebar_spacing: "rebar spacing", stirrup_spacing: "stirrup spacing", cover: "cover", thickness: "thickness",
  rebar_dia: "bar dia", size: "size", grade: "grade", bar_count: "bar count",
};
export const attrLabel = (a?: string) => (a ? ATTR_HI[a] ?? a.replace(/_/g, " ") : "");
const ACT_HI: Record<string, string> = { hot_work: "hot work", pour: "dhalai", height: "height work", excavation: "khudai" };
export const levelLabel = (l?: string | null) => (l ? l.replace(/^L(\d+)$/, "Level $1") : "");

export function locLabel(c: CheckResult, s: Slots): string {
  const loc = c.location;
  if (loc) {
    const head = loc.grid && loc.grid !== "Slab" ? loc.grid : loc.zone ?? (loc.grid === "Slab" ? "Slab" : loc.location_id);
    return loc.grid === "Slab" ? `${levelLabel(loc.level)} slab` : `${head}, ${levelLabel(loc.level)}`;
  }
  return [s.grid?.value ?? s.zone?.value, levelLabel(s.level?.value)].filter(Boolean).join(", ");
}

/** Make a display string speakable. */
export function toSpeech(t: string): string {
  return t
    .replace(/(\d+(?:\.\d+)?)\s?mm\b/g, "$1 millimeter")
    .replace(/±\s?(\d+)/g, "plus minus $1")
    .replace(/\bRFI-0*(\d+)/g, (_m, n) => `R F I ${n}`)
    .replace(/\bNCR\b/g, "N C R")
    .replace(/\bOBS-0*(\d+)/g, (_m, n) => `observation number ${n}`)
    .replace(/\bHWP-0*(\d+)/g, (_m, n) => `H W P ${n}`)
    .replace(/\b([A-Z])-(\d{3})\b/g, "$1 $2")
    .replace(/\b([A-H])-(\d{1,2})\b/g, "$1 $2")
    .replace(/\bR(\d+)\b/g, "revision $1")
    .replace(/\bL(\d+)\b/g, "level $1")
    .replace(/QC-[A-Z]+-[A-Z]+-\d+/g, "checklist")
    .replace(/\bCL-[A-Z0-9-]+/g, "")
    .replace(/—/g, ", ")
    .replace(/\s+/g, " ")
    .trim();
}

export interface Reply { text: string; speech: string }
const R = (text: string): Reply => ({ text, speech: toSpeech(text) });

export function challengeReply(c: CheckResult, s: Slots, prefix = ""): Reply {
  const parts: string[] = [];
  if (prefix) parts.push(prefix);
  const loc = locLabel(c, s);
  const attr = attrLabel(c.attribute ?? s.attribute?.value);
  for (const k of c.contradictions) parts.push(describe(k, c, s, loc, attr));
  const hasDim = c.contradictions.some((k) => k.kind === "dimension_mismatch");
  parts.push(hasDim
    ? "Kya aap observation log karna chahte hain, RFI raise karna chahte hain, ya NCR?"
    : "Kya aap observation log karna chahte hain, ya RFI raise karna chahte hain?");
  return R(parts.join(" "));
}

function describe(k: Contradiction, c: CheckResult, s: Slots, loc: string, attr: string): string {
  const e = k.evidence;
  switch (k.kind) {
    case "revision_mismatch": {
      let t = `Drawing ${e.drawing_number} ki latest revision ${e.revision} hai, jo ${spokenDate(e.issued_on)} ko issue hui thi.`;
      if (e.expected != null) t += ` Usme ${loc} ki ${attr} ${e.expected} ${e.unit ?? "mm"} hai.`;
      if (e.via_rfi) t += ` ${e.via_rfi} ne yeh confirm kiya tha.`;
      if (e.claimed_revision) t += ` ${e.claimed_revision} superseded hai${e.claimed_revision_value != null ? `, usme ${e.claimed_revision_value} tha` : ""}.`;
      if (c.value != null && e.expected != null && !c.contradictions.some((x) => x.kind === "dimension_mismatch")) t += ` Aapka ${c.value} ${c.unit ?? "mm"} latest drawing se match karta hai.`;
      return t;
    }
    case "dimension_mismatch": {
      if (e.claimed != null && c.value != null && e.claimed === c.value)
        return `Dhyan dijiye: ${loc} pe aapne ${attr} ${c.value} ${e.unit ?? "mm"} bola, lekin ${e.drawing_number} ${e.revision} mein ${e.expected} ${e.unit ?? "mm"} hai, tolerance ±${e.tolerance ?? 0}. Yeh ${Math.abs(Number(e.claimed) - Number(e.expected))} ${e.unit ?? "mm"} ka farak hai.${e.code_ref ? ` Reference ${e.code_ref}.` : ""}`;
      return `Aapne drawing mein ${e.claimed} bataya, lekin ${e.drawing_number} ${e.revision} mein ${e.expected} ${e.unit ?? "mm"} hai.`;
    }
    case "unknown_drawing":
      if (e.drawing_id) return `Drawing ${e.drawing_number} is location ke liye applicable nahi hai; ${e.drawing_id.replace("@", " ")} lagu hai, issue ${spokenDate(e.issued_on)}.`;
      return `Drawing ${e.drawing_number} register mein nahi mili.`;
    default:
      return k.detail;
  }
}

export function unknownDrawingReply(k: Contradiction): Reply {
  const cands = (k.evidence.candidates ?? []).slice(0, 4).join(", ");
  return R(`Drawing ${k.evidence.drawing_number} register mein nahi mili. Kripya drawing number dobara bataiye${cands ? ` — register mein ${cands} hain` : ""}. Tab tak kuch log nahi hoga.`);
}

export function missingReply(c: CheckResult, s: Slots): Reply {
  const m = c.missing[0];
  const f = m.evidence.field;
  if (f === "level") {
    const levels = (m.evidence.candidates ?? []).map((id) => levelLabel(id.split(":")[2])).join(" ya ");
    return R(`${s.grid?.value ?? s.zone?.value} kaunse level pe hai — ${levels}?`);
  }
  if (f === "element") return R(`${locLabel(c, s)} pe kaunsa element — ${(m.evidence.candidates ?? []).join(" ya ")}?`);
  if (f === "location_unknown") return R(`${m.evidence.claimed} project register mein nahi mila. Grid line dobara bataiye.`);
  if (f === "location") return R("Location bataiye — kaunsi grid line aur level? Jaise C-5, Level 3.");
  if (f === "attribute") return R(`${locLabel(c, s) || "Wahan"} pe kya check kiya — spacing, cover ya thickness?`);
  if (f === "value") return R(`${attrLabel(c.attribute ?? s.attribute?.value) || "Value"} kitni hai? mm mein bataiye.`);
  return R("Thoda aur detail chahiye. Location, attribute aur value bataiye.");
}

const SLOT_SAY: Partial<Record<SlotName, (s: Slots) => string>> = {
  grid: (s) => `grid ${s.grid?.value}`,
  level: (s) => levelLabel(s.level?.value),
  zone: (s) => `${s.zone?.value}`,
  attribute: (s) => attrLabel(s.attribute?.value),
  value: (s) => `${s.value?.value} ${s.unit?.value ?? "mm"}`,
  drawingNumber: (s) => `drawing ${s.drawingNumber?.value}`,
  element: (s) => `${s.element?.value}`,
  activity: (s) => ACT_HI[s.activity?.value ?? ""] ?? String(s.activity?.value),
};
export function confirmSlotsReply(names: SlotName[], s: Slots, resolvedDrawing?: string): Reply {
  const said = names.map((n) => (n === "drawingNumber" && resolvedDrawing ? `drawing ${resolvedDrawing}` : SLOT_SAY[n]?.(s) ?? n)).join(", ");
  return R(`Awaaz saaf nahi thi. Maine suna: ${said}. Kya yeh sahi hai? Haan ya nahi boliye.`);
}

/** "Fire watcher assigned, distinct from welder (..)" → "Fire watcher assigned"; max 2 items + count. */
function shortList(items?: string[]): string {
  if (!items?.length) return "";
  const short = items.map((i) => i.split(/[,(—;]| - /)[0].trim());
  const head = short.slice(0, 2).join("; ");
  return items.length > 2 ? `${head}, aur ${items.length - 2} aur items` : head;
}

export function blockerReply(c: CheckResult, s: Slots, attempted?: string): Reply {
  const parts: string[] = [];
  if (attempted) parts.push(`${attempted} abhi allowed nahi hai.`);
  parts.push("Ruko.");
  const loc = locLabel(c, s);
  for (const b of c.blockers) {
    if (b.kind === "permit_blocker") {
      if (b.evidence.permit_id) parts.push(`${loc} pe ${ACT_HI[s.activity?.value ?? ""] ?? "kaam"} permit ${b.evidence.permit_id} active hai, lekin mandatory check pending hai: ${shortList(b.evidence.checks)}. Jab tak yeh confirm na ho, ${ACT_HI[s.activity?.value ?? ""] ?? "kaam"} shuru nahi ho sakta.`);
      else parts.push(`${loc} pe ${ACT_HI[s.activity?.value ?? ""] ?? "is kaam"} ke liye koi active permit nahi hai.`);
    } else if (b.kind === "hold_point_blocker") {
      parts.push(`${loc} ka pre-pour checklist ${b.evidence.checklist} ka hold point release nahi hua: ${shortList(b.evidence.items)}. Dhalai shuru nahi ho sakti.`);
    }
  }
  parts.push("Kaam rokna hai, NCR raise karna hai, ya cancel?");
  return R(parts.join(" "));
}

export function readbackReply(c: CheckResult, s: Slots, prefix = ""): Reply {
  const loc = locLabel(c, s);
  const d = c.drawing ? `${c.drawing.drawing_number} ${c.drawing.revision}` : null;
  let t = prefix ? prefix + " " : "";
  if (c.mode === "defect") t += `${loc} pe ${s.element?.value ?? ""} ${s.defect?.value}${d ? `, drawing ${d} se link hoga` : ""}.`;
  else if (c.mode === "activity") t += `${loc} pe ${ACT_HI[s.activity?.value ?? ""]} ke liye koi blocker nahi mila.`;
  else t += `${loc}, ${c.element ?? s.element?.value ?? ""} ${attrLabel(c.attribute)} ${c.value} ${c.unit ?? "mm"}${c.fact && d ? ` — ${d} ke hisaab se sahi hai` : d ? ` — drawing ${d}` : ""}.`;
  t = t.replace(/\s+/g, " ").replace(" ,", ",");
  return R(`${t} Log kar doon?`);
}

export function doneReply(decision: string, obsId?: string, rfiId?: string, drawingLabel?: string): Reply {
  switch (decision) {
    case "log_observation": return R(`Observation ${obsId} log ho gaya${drawingLabel ? `, drawing ${drawingLabel} ke saath link kiya` : ""}.`);
    case "raise_rfi": return R(`${rfiId} raise ho gaya, status Open. Observation ${obsId} usse link kar diya.`);
    case "raise_ncr": return R(`NCR ke liye observation ${obsId} log ho gaya. QA team ko notify kijiye.`);
    case "stop_work": return R(`Kaam rokne ka order ${obsId} log ho gaya. Safety officer ko turant inform kijiye.`);
    case "cancel": return R("Theek hai, cancel kar diya. Kuch log nahi hua.");
    default: return R("Theek hai.");
  }
}

export const REFUSE = {
  confirmFirst: "Pehle confirm kijiye, phir log karunga.",
  clarifyFirst: "Log karne se pehle ek baat:",
  ambiguousYes: "Log karna hai ya RFI raise karna hai? Saaf boliye.",
  rfiDeclined: "Theek hai, RFI nahi. Toh observation log kar doon?",
  nothing: "Main sun raha hoon. Location, attribute aur value boliye — jaise, C-5 pe spacing 180 mm.",
};
export { R as reply };
